extensions/iceberg/s3/src/main/java/io/deephaven/iceberg/util/IcebergToolsS3.java

```diff
@@ -102,4 +110,40 @@ public static IcebergCatalogAdapter createGlue(
 
         return new IcebergCatalogAdapter(catalog, properties);
     }
+
+    /**
+     * Create an Iceberg catalog adapter. Instead of the AWS clients being configured via the various Iceberg-specific
+     * properties (found amongst {@link S3FileIOProperties}, {@link HttpClientProperties}, and {@link AwsProperties}),
+     * the clients are created in the same way that Deephaven's AWS clients are configured with respect to
+     * {@code instructions}. This ensures, amongst other things, that Iceberg's AWS configuration and credentials are
+     * in-sync with Deephaven's AWS configuration and credentials for S3 access. The {@code instructions} will
+     * automatically be used as special instructions if {@link IcebergInstructions#dataInstructions()} if not explicitly
+     * set.
+     *
+     * <p>
+     * The caller is still responsible for providing the properties necessary as specified in
+     * {@link IcebergTools#createAdapter(String, Map, Map)}.
+     *
+     * @param name the name of the catalog; if omitted, the catalog URI will be used to generate a name
+     * @param properties a map containing the Iceberg catalog properties to use
+     * @param hadoopConfig a map containing Hadoop configuration properties to use
+     * @param instructions the s3 instructions
+     * @return the Iceberg catalog adapter
+     *
+     * @see DeephavenAwsClientFactory#addToProperties(S3Instructions, Map)
+     */
+    public static IcebergCatalogAdapter createAdapter(
+            @Nullable final String name,
+            @NotNull final Map<String, String> properties,
+            @NotNull final Map<String, String> hadoopConfig,
```

Comment:
Is `hadoopConfig` used in an S3-backed adapter with S3 file IO? Can we drop this as a parameter and create an empty config internal to this function?

Reply:
It's a good question; `org.apache.iceberg.aws.s3.S3FileIO` does _not_ use hadoop conf. That said, this method is _not_ imposing `S3FileIO`. It looks like GlueCatalog is technically written in a way where hadoopConf can be passed along if something besides S3FileIO is used... maybe it's possible to use GlueCatalog and _not_ use S3 as the warehouse... for example, maybe some sort of other AWS NFS storage, I'm not sure. I'm going to add more explicit documentation about this.